`wait_container_is_ready` throwing TimeoutException on first try

After updating to v3.5.0 our tests using `PostgresContainer`s are failing.

My guess is that this has to do with the changes regarding transient exceptions. We use `psycopg2` and the error thrown on unsuccessful connection is not inheriting `sqlalchemy.exc.OperationalError` but rather is a plain `AttributeError` with message:

```'NoneType' object has no attribute '_stan_scope'```

With sqlalchemy and asyncio I, too get an `ConnectionError`:

```
ERROR [  1%]{"asctime": "2022-04-04 23:27:09,628", "name": "testcontainers.core.container", "lineno": 51, "levelname": "INFO", "message": "Pulling image postgres:11", "workspace": null, "region": "westeurope", "instana_trace_id": "NO_CONTEXT"}
{"asctime": "2022-04-04 23:27:10,332", "name": "testcontainers.core.container", "lineno": 62, "levelname": "INFO", "message": "Container started: 7b427a53a0", "workspace": null, "region": "westeurope", "instana_trace_id": "NO_CONTEXT"}
{"asctime": "2022-04-04 23:27:10,332", "name": "testcontainers.core.waiting_utils", "lineno": 46, "levelname": "INFO", "message": "Waiting to be ready...", "workspace": null, "region": "westeurope", "instana_trace_id": "NO_CONTEXT"}
Pulling image postgres:11
Container started: 7b427a53a0
Waiting to be ready...

test setup failed
pytestconfig = <_pytest.config.Config object at 0x10fb6cb20>

    @pytest.fixture(scope="session")
    def async_db_container(pytestconfig: Any) -> AsyncPostgresContainer:
>       with _create_container(
            "postgres:11",
            AsyncPostgresContainer,
        ) as container:

conftest.py:45: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../.venv/lib/python3.10/site-packages/testcontainers/core/container.py:69: in __enter__
    return self.start()
../.venv/lib/python3.10/site-packages/testcontainers/core/generic.py:55: in start
    self._connect()
../.venv/lib/python3.10/site-packages/testcontainers/core/waiting_utils.py:49: in wrapper
    return wrapped(*args, **kwargs)
conftest.py:40: in _connect
    asyncio.run(self._test_connection(engine))
../.venv/lib/python3.10/site-packages/nest_asyncio.py:32: in run
    return loop.run_until_complete(future)
../.venv/lib/python3.10/site-packages/nest_asyncio.py:70: in run_until_complete
    return f.result()
/usr/local/Cellar/python@3.10/3.10.2/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/futures.py:201: in result
    raise self._exception
/usr/local/Cellar/python@3.10/3.10.2/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/tasks.py:232: in __step
    result = coro.send(None)
conftest.py:35: in _test_connection
    await engine.connect()
../.venv/lib/python3.10/site-packages/sqlalchemy/ext/asyncio/engine.py:155: in start
    await (greenlet_spawn(self.sync_engine.connect))
../.venv/lib/python3.10/site-packages/sqlalchemy/util/_concurrency_py3k.py:134: in greenlet_spawn
    result = context.throw(*sys.exc_info())
../.venv/lib/python3.10/site-packages/sqlalchemy/future/engine.py:406: in connect
    return super(Engine, self).connect()
../.venv/lib/python3.10/site-packages/sqlalchemy/engine/base.py:3234: in connect
    return self._connection_cls(self, close_with_result=close_with_result)
../.venv/lib/python3.10/site-packages/sqlalchemy/engine/base.py:96: in __init__
    else engine.raw_connection()
../.venv/lib/python3.10/site-packages/sqlalchemy/engine/base.py:3313: in raw_connection
    return self._wrap_pool_connect(self.pool.connect, _connection)
../.venv/lib/python3.10/site-packages/sqlalchemy/engine/base.py:3280: in _wrap_pool_connect
    return fn()
../.venv/lib/python3.10/site-packages/sqlalchemy/pool/base.py:310: in connect
    return _ConnectionFairy._checkout(self)
../.venv/lib/python3.10/site-packages/sqlalchemy/pool/base.py:868: in _checkout
    fairy = _ConnectionRecord.checkout(pool)
../.venv/lib/python3.10/site-packages/sqlalchemy/pool/base.py:476: in checkout
    rec = pool._do_get()
../.venv/lib/python3.10/site-packages/sqlalchemy/pool/impl.py:145: in _do_get
    with util.safe_reraise():
../.venv/lib/python3.10/site-packages/sqlalchemy/util/langhelpers.py:70: in __exit__
    compat.raise_(
../.venv/lib/python3.10/site-packages/sqlalchemy/util/compat.py:207: in raise_
    raise exception
../.venv/lib/python3.10/site-packages/sqlalchemy/pool/impl.py:143: in _do_get
    return self._create_connection()
../.venv/lib/python3.10/site-packages/sqlalchemy/pool/base.py:256: in _create_connection
    return _ConnectionRecord(self)
../.venv/lib/python3.10/site-packages/sqlalchemy/pool/base.py:371: in __init__
    self.__connect()
../.venv/lib/python3.10/site-packages/sqlalchemy/pool/base.py:665: in __connect
    with util.safe_reraise():
../.venv/lib/python3.10/site-packages/sqlalchemy/util/langhelpers.py:70: in __exit__
    compat.raise_(
../.venv/lib/python3.10/site-packages/sqlalchemy/util/compat.py:207: in raise_
    raise exception
../.venv/lib/python3.10/site-packages/sqlalchemy/pool/base.py:661: in __connect
    self.dbapi_connection = connection = pool._invoke_creator(self)
../.venv/lib/python3.10/site-packages/sqlalchemy/engine/create.py:590: in connect
    return dialect.connect(*cargs, **cparams)
../.venv/lib/python3.10/site-packages/sqlalchemy/engine/default.py:597: in connect
    return self.dbapi.connect(*cargs, **cparams)
../.venv/lib/python3.10/site-packages/sqlalchemy/dialects/postgresql/asyncpg.py:777: in connect
    await_only(self.asyncpg.connect(*arg, **kw)),
../.venv/lib/python3.10/site-packages/sqlalchemy/util/_concurrency_py3k.py:76: in await_only
    return current.driver.switch(awaitable)
../.venv/lib/python3.10/site-packages/sqlalchemy/util/_concurrency_py3k.py:129: in greenlet_spawn
    value = await result
../.venv/lib/python3.10/site-packages/asyncpg/connection.py:2085: in connect
    return await connect_utils._connect(
../.venv/lib/python3.10/site-packages/asyncpg/connect_utils.py:895: in _connect
    raise last_error
../.venv/lib/python3.10/site-packages/asyncpg/connect_utils.py:881: in _connect
    return await _connect_addr(
../.venv/lib/python3.10/site-packages/asyncpg/connect_utils.py:781: in _connect_addr
    return await __connect_addr(params, timeout, True, *args)
../.venv/lib/python3.10/site-packages/asyncpg/connect_utils.py:825: in __connect_addr
    tr, pr = await compat.wait_for(connector, timeout=timeout)
../.venv/lib/python3.10/site-packages/asyncpg/compat.py:66: in wait_for
    return await asyncio.wait_for(fut, timeout)
/usr/local/Cellar/python@3.10/3.10.2/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/tasks.py:445: in wait_for
    return fut.result()
/usr/local/Cellar/python@3.10/3.10.2/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/futures.py:201: in result
    raise self._exception
/usr/local/Cellar/python@3.10/3.10.2/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/tasks.py:234: in __step
    result = coro.throw(exc)
../.venv/lib/python3.10/site-packages/asyncpg/connect_utils.py:699: in _create_ssl_connection
    do_ssl_upgrade = await pr.on_data
/usr/local/Cellar/python@3.10/3.10.2/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/futures.py:284: in __await__
    yield self  # This tells Task to wait for completion.
/usr/local/Cellar/python@3.10/3.10.2/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/tasks.py:304: in __wakeup
    future.result()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <Future finished exception=ConnectionError('unexpected connection_lost() call')>

    def result(self):
        """Return the result this future represents.
    
        If the future has been cancelled, raises CancelledError.  If the
        future's result isn't yet available, raises InvalidStateError.  If
        the future is done and has an exception set, this exception is raised.
        """
        if self._state == _CANCELLED:
            exc = self._make_cancelled_error()
            raise exc
        if self._state != _FINISHED:
            raise exceptions.InvalidStateError('Result is not ready.')
        self.__log_traceback = False
        if self._exception is not None:
>           raise self._exception
E           ConnectionError: unexpected connection_lost() call

/usr/local/Cellar/python@3.10/3.10.2/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/futures.py:201: ConnectionError
```

--- a/testcontainers/core/waiting_utils.py
+++ b/testcontainers/core/waiting_utils.py
@@ -26,7 +26,7 @@ logger = setup_logger(__name__)
 
 
 # Get a tuple of transient exceptions for which we'll retry. Other exceptions will be raised.
-TRANSIENT_EXCEPTIONS = (TimeoutError, ConnectionResetError, BrokenPipeError)
+TRANSIENT_EXCEPTIONS = (TimeoutError, ConnectionError)
 
 
 def wait_container_is_ready(*transient_exceptions):

